base/tps/src/org/dogtagpki/server/tps/rest/ActivityService.java

```diff
@@ -136,25 +140,46 @@ protected void retrieveActivitiesWithVLV(
             Integer size,
             ActivityCollection response) throws Exception {
 
+        String method = "ActivityService.retrieveActivitiesWithVLV: ";
+	CMS.debug(method);
         // search with VLV sorted by date in reverse order
         IDBVirtualList<ActivityRecord> list = database.findRecords(
                 null, null, new String[] { "-date" }, size);
 
+        List<String> authorizedProfiles = getAuthorizedProfiles();
+
         int total = list.getSize();
+        int retTotal = 0;
 
         // return entries in the requested page
-        for (int i = start; i < start + size && i < total; i++) {
-            ActivityRecord record = list.getElementAt(i);
-
-            if (record == null) {
-                CMS.debug("ActivityService: Activity record not found");
-                throw new PKIException("Activity record not found");
+        if (authorizedProfiles != null) {
+            if (authorizedProfiles.contains(UserResource.ALL_PROFILES)) {
+                for (int i = start; i < start + size && i < total; i++) {
+                    ActivityRecord record = list.getElementAt(i);
+
+                    response.addEntry(createActivityData(record));
+                }
+                retTotal = total;
+            } else { // not authorized for all profiles
+                for (int i = start; i < start + size && i < total; i++) {
```

Comment:
This loop is actually used to page the results based on the starting index and page size parameters, so if there are no profile restrictions the method should return a constant number of records from index `<start>` to `<start + size - 1>`. However, if there are profile restrictions, the number of records per page will not be constant anymore, so I'm not sure how the `start` and `size` parameters should be interpreted in this case. It's possible that some pages will be empty, which might confuse the user.

Reply:
yeah this question occurred to me as well, but I don't have an easy solution for this.  Could we maybe create a "restricted" record filled with something like "<restricted>" for each restricted record, and return them along with the allowed ones?